### PR TITLE
Add option to allow console redirect in ESXi bootstrap

### PR DESCRIPTION
**Pull request by nortonluo**
In many rack mount servers or customized servers, there is no monitor connected, UART port are usually used as console. VMware ESXi use VGA card as console. User need pass VM kernel To support this use, I add three options: "gdbPort","logPort","debugLogToSerial". 
[VMware ESXi reference](https://pubs.vmware.com/vsphere-50/index.jsp?topic=%2Fcom.vmware.vsphere.install.doc_50%2FGUID-C65306C0-DA37-4F45-8A50-31F8D109BB1D.html])

To support this , 'on-http/data/templates/esx-boot-cfg', 'on-http/data/templates/esx-ks'(This PR) and 'on-tasks/lib/task-data/tasks/install-esx.js'(in another PR for on-tasks) are changed.
The default value of those options keep console in VGA as it does before. User can customize the console redirect setting in the OS bootstrapping payload.
Jenkins depends on https://github.com/RackHD/on-http/pull/636

### CODE REVIEW

**Comment by nortonluo** [comment]
@pengz1 @iceiilin @anhou @yyscamper

File: lib/task-data/tasks/install-esx.js [hunk 1]
```diff
@@ -17,6 +17,9 @@ module.exports = {
         esxBootConfigTemplateUri: '{{ api.templates }}/{{ options.esxBootConfigTemplate }}?nodeId={{ task.nodeId }}', //jshint ignore: line
         comport: 'com1',
         comportaddress: '0x3f8', //com1=0x3f8, com2=0x2f8, com3=0x3e8, com4=0x2e8
+        gdbPort: 'default',
+        logPort: 'com1',
+        debugLogToSerial: '1',
```
Comment (yyscamper) [inline lib/task-data/tasks/install-esx.js]: This should be changed to `boolean`

**Comment by yyscamper** [comment]
@nortonluo : Please also update the schema file: https://github.com/RackHD/on-tasks/blob/master/lib/task-data/schemas/install-esxi.json and its unit-test: https://github.com/RackHD/on-tasks/blob/master/spec/lib/task-data/schemas/install-esxi-spec.js

**Comment by iceiilin** [comment]
Will these default settings misbehave in other HW platforms when installing ESXi?

File: lib/task-data/tasks/install-esx.js [hunk 2]
```diff
@@ -17,6 +17,9 @@ module.exports = {
         esxBootConfigTemplateUri: '{{ api.templates }}/{{ options.esxBootConfigTemplate }}?nodeId={{ task.nodeId }}', //jshint ignore: line
         comport: 'com1',
         comportaddress: '0x3f8', //com1=0x3f8, com2=0x2f8, com3=0x3e8, com4=0x2e8
+        gdbPort: 'default',
+        logPort: 'com1',
```
Comment (nortonluo) [inline lib/task-data/tasks/install-esx.js]: comport is the device name of UART in ESXi. comportaddress is the physical address of UART. logport is the port that ESXi throw out debug log during boot/running. There are different. I think they couldn't get merged

File: lib/task-data/tasks/install-esx.js (hunk 1 above)
Comment (nortonluo) [inline lib/task-data/tasks/install-esx.js]: This variable is a VMkernel value. It has value 0,1,2 so far.

**Comment by nortonluo** [comment]
@iceiilin Those parameters are kept in ESXi default value by default. That is the same as the value before this PR. I Just test this (change parameters or not) on Rinjin, it is OK.

**Comment by iceiilin** [comment]
Please also update schema file here (not only schema unit test) https://github.com/RackHD/on-tasks/blob/master/lib/task-data/schemas/install-esxi.json

File: lib/task-data/schemas/install-esxi.json [hunk 3]
```diff
@@ -55,6 +55,32 @@
                 "0x3e8",
                 "0x2e8"
             ]
+        },
+        "gdbPort":{
+            "description":"GDB debug serial port mode",
```
Comment (pengz1) [inline lib/task-data/schemas/install-esxi.json]: "mode" is unecessary

**Comment by pengz1** [comment]
Please make sure to update document in:
http://rackhd.readthedocs.io/en/latest/rackhd/install_os.html

**Pull request merged**